File: python/models/profesor.py
from .persona import Persona


class Profesor(Persona):

    def __init__(self, cedula, nombre_completo, correo,
                 especialidad, materia_asignada):

        super().__init__(cedula, nombre_completo, correo)

        self.especialidad = especialidad
        self.materia_asignada = materia_asignada

    def mostrar_informacion(self):
        return (
            f"{self.mostrar_datos()}\n"
            f"Especialidad: {self.especialidad}\n"
            f"Materia: {self.materia_asignada}"
        )

File: java/Profesor.java
public class Profesor extends Persona {
    private final String especialidad;
    private final String materia;

    public Profesor(String cedula, String nombre, String correo, String especialidad, String materia) {
        super(cedula, nombre, correo);
        this.especialidad = especialidad;
        this.materia = materia;
    }

    public String getEspecialidad() {
        return especialidad;
    }

    public String getMateria() {
        return materia;
    }

    public String toFileLine() {
        return String.join(",", getCedula(), getNombre(), getCorreo(), especialidad, materia);
    }
}
